jest-puppeteer.config.js

```diff
@@ -1,5 +1,14 @@
 module.exports = {
   launch: {
-    headless: process.env.HEADLESS !== 'false'
-  },
+    devtools: (({ devtools, iframes }) => {
+      if (devtools) {
+        iframes && console.warn('You cannot enable devtools while testing a UI with iframes in headful mode')
+        return !iframes
+      }
+    })({ devtools: process.env.DEVTOOLS === 'true', iframes: process.env.IFRAMES === 'true' }),
+    headless: process.env.HEADLESS !== 'false',
+    // Workaround for issue when accessing cross domain iframes in puppeteer while not headless:
+    // https://github.com/GoogleChrome/puppeteer/issues/4960
+    args: process.env.HEADLESS === 'false' && process.env.IFRAMES === 'true' ? ['--disable-features=site-per-process'] : []
```

Comment:
Since we have tests with iframes, doesn't this always need to be true?

Reply:
This one is an annoyance of puppeteer. The flag is only needed when running this in headful mode (e.g. debugging). The tests run fine and can find frames in headless mode without the flag. Once in headful mode puppeteer will not see the frames.

I don't want to enable the flag all the time as I don't know what other side effects may occur when this is turned on - I'd rather run the tests with no flags enabled when running in an automated fashion.

I set it as a flag separate from headless because I want our parameters to be explicit. I wouldn't want someone's tests to fail while debugging because HEADLESS was set to false and they did not realize HEADLESS sets a second flag as well. My concern is that while this flag fixes iframe issues, it could potentially have some other unknown effect that could cause another issue.

Reply:
Yes. I may be overly paranoid here. I can tie this flag into the HEADLESS option ( I did that initially)  - I just don't want that to cause test failures for someone in the future for other reasons. The headless / iframe issue was annoying for me, I would like to guard against it for future devs. 

I could also issue another log statement warning about the iframe issue.